src/canvas/canvas-tag.service.ts

```diff
@@ -0,0 +1,84 @@
+import { ConflictException, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { CanvasTagEntity } from './entity/canvas-tag.entity';
+import { Repository } from 'typeorm';
+import {
+  CanvasTagCreateCommand,
+  CanvasTagDeleteCommand,
+  CanvasTagUpdateCommand,
+} from './canvas.interface';
+import {
+  ListFilter,
+  PageableUtils,
+  PagedResult,
+} from '../common/pageable.utils';
+import { Uuid } from '../common/common.interface';
+
+@Injectable()
+export class CanvasTagService {
+  constructor(
+    @InjectRepository(CanvasTagEntity)
+    private readonly canvasTagRepository: Repository<CanvasTagEntity>,
+  ) {}
+
+  public async create(
+    command: CanvasTagCreateCommand,
+  ): Promise<CanvasTagEntity> {
+    const tagName = command.name?.trim().toUpperCase();
+    const existingTag = await this.canvasTagRepository.findOne({
```

Comment:
If we treat tags names as unique maybe we should create unique index on the DB level (for name column)

Reply:
Done